pkg/gorilla/gorilla.go

```diff
@@ -159,15 +168,17 @@ func (ws *WebSocket) Send(method string, params []interface{}) (interface{}, err
 	select {
 	case <-timeout:
 		return nil, ErrTimeout
-	case res := <-responseChan:
-		if res.ID != id {
+	case raw, open := <-responseChan:
+		if !open {
+			return nil, errors.New("channel closed")
+		}
+		res, ok := raw.(rpc.RPCResponse)
```

Comment:
In this PR we talked into this too but the thing is even if we do the user couldn't use it because the RPC module is internal
if we want to achieve that we need to make rpc package public as well for me we can expose some rpc parts to improve the developer experience.

Reply:
The intention is to allow it to work in embedded mode, and we shouldn't be exposing the implementation details like rpc.